### PR TITLE
DATAUP-643 multiple parameter set warning

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

This adds a warning banner to the top of bulk import cells if an XSV file has multiple non-unique, non-file parameters. For example, if an assembly file looks like:

```
FASTA file path, Assembly object name, Assembly Type, Minimum contig length
first_file.fasta,  first_object,  draft isolate, 1000
second_file.fasta, second_object, draft isolate, 1000
```
there would be no warning, as the parameters that aren't input files or output names are identical across all rows. If they were not identical, then the first row's params would be used, and there would be a yellow warning banner:
Screen Shot 2022-02-28 at 10 57 41 PM

I also found a bug where the widgets in the configure tab weren't being unloaded properly, which might cause memory issues, especially with large amounts of rows.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-643
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
* Use import spec files with: 
* 1 row - no warning
* multiple rows of unique params - no warning
* multiple rows of different params - warning
* mix and match with different types - warning should stick to only the appropriate file type tab
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [x] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: nbextensions/bulkImportCell/tabs/configure.js [hunk 1]
```diff
@@ -147,16 +151,68 @@ define([
             const appSpec = specs[typesToFiles[selectedFileType].appId];
             const filePathNode = ui.getElement('input-container.file-paths');
             const paramNode = ui.getElement('input-container.params');
+            showConfigMessage();
             return Promise.all([
                 buildFilePathWidget(filePathNode, appSpec).then((instance) => {
-                    filePathWidget = instance.widget;
+                    filePathWidget = instance.instance;
```
Comment (briehl) [inline nbextensions/bulkImportCell/tabs/configure.js:157]: This was a little annoying, just a name change and now the filePathWidget and paramsWidget get stopped and removed as usual. They were being set as `undefined` before, and the stop code tests for existence first, so no errors. Yay for code coverage, I guess.

File: nbextensions/bulkImportCell/tabs/configure.js [hunk 2]
```diff
@@ -147,16 +151,68 @@ define([
             const appSpec = specs[typesToFiles[selectedFileType].appId];
             const filePathNode = ui.getElement('input-container.file-paths');
             const paramNode = ui.getElement('input-container.params');
+            showConfigMessage();
```
Comment (briehl) [inline nbextensions/bulkImportCell/tabs/configure.js:154]: Show the message here - it gets rendered every time the widgets get rendered, so we change it with the different file types.

File: test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js [hunk 3]
```diff
@@ -219,49 +219,6 @@ define([
                 ]);
             });
 
-            it('should build a bulk import cell with appParameters premade info', () => {
```
Comment (briehl) [inline test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js:222]: Moved down.

File: test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js [hunk 4]
```diff
@@ -304,6 +261,178 @@ define([
                     })
                 ).toThrow();
             });
+
+            describe('xsv specific setup', () => {
+                const inFiles = ['dvh.fasta', 'styphi.fasta', 'mex.fasta'],
```
Comment (briehl) [inline test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js:266]: Mostly the same as the moved test, just added multiple cases to check that file order is maintained, and the warning message shows up as expected.

File: test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js [hunk 5]
```diff
@@ -470,7 +599,7 @@ define([
                             });
                         }
                     );
-                    expect(console.error.calls.allArgs()).toEqual([['running execute!']]);
+                    expect(console.error.calls.allArgs()).toContain(['running execute!']);
```
Comment (briehl) [inline test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js:519]: There's some kind of timing issue here where the (now fixed) configure tab tries to remove uninitialized widgets on change. The "error" doesn't do anything, but it does make a `console.error` call. This just focuses on the one we want to see.

**Review by ialarmedalien — APPROVED** [review]
Looks good!

**Pull request merged**